source/core/ib_particle.cc

```diff
@@ -43,27 +44,56 @@ IBParticle<dim>::initialise_all()
       position[2] = 0;
     }
 
-  last_forces           = forces;
-  last_position         = position;
-  last_velocity         = velocity;
-  velocity_iter         = velocity;
-  last_omega            = omega;
-  omega_iter            = omega;
-  last_angular_position = angular_position;
+  last_forces   = forces;
+  last_torques  = torques;
+  velocity_iter = velocity;
+
+  omega_iter           = omega;
+  omega_impulsion      = 0;
+  omega_impulsion_iter = 0;
+  impulsion            = 0;
+  impulsion_iter       = 0;
+  contact_impulsion    = 0;
+
+  last_position.resize(3);
+  last_velocity.resize(3);
+  last_angular_position.resize(3);
+  last_omega.resize(3);
+
+  for (unsigned int i = 0; i < 3; ++i)
+    {
+      last_position[i]         = position;
+      last_velocity[i]         = velocity;
+      last_angular_position[i] = angular_position;
+      last_omega[i]            = omega;
+    }
```

Comment:
I don't understand.
You loop from 0 to 3, but last_position, last_velocity, etc. I thought they were Tensor<1,dim> . You will be out of bound...
The only vector that is always size 3 is angular position and omega